OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -326,6 +352,9 @@ public override Activity Tick(Actor self)
 					moveFraction = MoveFractionTotal;
 				UpdateCenterLocation(self, Move.mobile);
 
+				if (IsCanceled)
+					ret.NextActivity = Move.NextActivity;
+
```

Comment:
The move code is subtle and extremely fragile. Please try very hard to not change the way it queues and runs activities.